test/unit/parachain/loans.test.ts

```diff
@@ -128,4 +134,173 @@ describe("DefaultLoansAPI", () => {
+
+            const totalCollateralThresholdAdjustedCollateralAmountBtc = governanceCurrencyCollateralBtc
+                .mul(collateralThresholdGovernanceCurrency)
+                .add(wrappedCurrencyCollateralBtc.mul(collateralThresholdWrappedCurrency));
+            const totalLiquidationThresholdAdjustedCollateralAmountBtc = governanceCurrencyCollateralBtc
+                .mul(liquidationThresholdGovernanceCurrency)
+                .add(wrappedCurrencyCollateralBtc.mul(liquidationThresholdWrappedCurrency));
+
+            const lendPositions = [
+                mockLendPosition(testGovernanceCurrency, governanceCurrencyLentAmount),
+                mockLendPosition(testRelayCurrency, relayCurrencyLentAmount, false),
+                mockLendPosition(wrappedCurrency, wrappedCurrencyLentAmount),
+            ];
+            const borrowPositions = [
+                mockBorrowPosition(testRelayCurrency, relayCurrencyBorrowedAmount),
+                mockBorrowPosition(testGovernanceCurrency, governanceCurrencyBorrowedAmount),
+            ];
+
+            const { ltv, collateralThresholdWeightedAverage, liquidationThresholdWeightedAverage } =
+                loansApi.getLoanCollateralInfo(lendPositions, borrowPositions, loanAssets);
+
+            const expectedLtv = totalBorrowedAmountBtc.div(totalCollateralAmountBtc);
+            const expectedAverageCollateralThreshold =
+                totalCollateralThresholdAdjustedCollateralAmountBtc.div(totalCollateralAmountBtc);
+            const expectedAverageLiquidationThreshold =
+                totalLiquidationThresholdAdjustedCollateralAmountBtc.div(totalCollateralAmountBtc);
+
+            expect(ltv.eq(expectedLtv), `LTV: ${ltv.toString()} does not match expected LTV: ${expectedLtv.toString()}`)
+                .to.be.true;
+            expect(
+                collateralThresholdWeightedAverage.eq(expectedAverageCollateralThreshold),
+                `Average collateral threshold: ${collateralThresholdWeightedAverage.toString()} does not match expected threshold: ${expectedAverageCollateralThreshold.toString()}`
+            ).to.be.true;
+            expect(
+                liquidationThresholdWeightedAverage.eq(expectedAverageLiquidationThreshold),
+                `Average liquidation threshold: ${liquidationThresholdWeightedAverage.toString()} does not match expected threshold: ${expectedAverageLiquidationThreshold.toString()}`
+            ).to.be.true;
+        });
+
+        it("should not throw when there are no positions", () => {
+            expect(loansApi.getLoanCollateralInfo([], [], loanAssets)).to.not.throw;
+        });
+
+        it("should not throw when there are no borrow positions", () => {
+            const lendPositions = [mockLendPosition(testGovernanceCurrency, 1)];
+            expect(loansApi.getLoanCollateralInfo(lendPositions, [], loanAssets)).to.not.throw;
+        });
```

Comment:
You can add another test that passes empty a non-empty `lendPositions`, a non-empty `borrowPositions`, and an empty `TickerToData` for `loanAssets` to make sure it doesn't throw

Reply:
Added test to ensure that it _throws_, see my response to the previous comment.